packages/airnode-node/src/evm/fulfillments/api-calls.test.ts

```diff
@@ -526,8 +240,854 @@ describe('submitApiCall', () => {
           `${RequestErrorMessage.ApiCallFailed} with error: Server did not respond`,
           txOpts
         );
+        expect(staticFulfillMock).not.toHaveBeenCalled();
+        expect(estimateFulfillMock).not.toHaveBeenCalled();
         expect(fulfillMock).not.toHaveBeenCalled();
       }
     );
   });
+
+  describe('testAndSubmitFulfill', () => {
+    describe('Pending API calls', () => {
+      test.each([gasTarget, gasTargetFallback])(
+        `successfully tests and submits a fulfill transaction for pending requests - %#`,
+        async (gasTarget) => {
+          const txOpts = { ...gasTarget, nonce: 5 };
+          const provider = new ethers.providers.JsonRpcProvider();
+          staticFulfillMock.mockResolvedValueOnce({ callSuccess: true, callData: '0x' });
+          fulfillMock.mockResolvedValueOnce({ hash: '0xtransactionId' });
+
+          const apiCall = fixtures.requests.buildSuccessfulApiCall({
+            id: '0xb56b66dc089eab3dc98672ea5e852488730a8f76621fd9ea719504ea205980f8',
+            data: {
+              encodedValue: '0x448b8ad3a330cf8f269f487881b59efff721b3dfa8e61f7c8fd2480389459ed3',
+              signature:
+                '0xda6d5aa27f48aa951ba401c8a779645f7d1fa4a46a5e99eb7da04b4e059449a834ca1058c85dfe8117305265228f8cf7ae64c3ef3c4d1cc191f77807227dac461b',
+            },
+            nonce: 5,
+          });
+          const [logs, err, data] = await apiCalls.testAndSubmitFulfill(createAirnodeRrpFake(), apiCall, {
+            gasTarget,
+            masterHDNode,
+            provider,
+          });
+          expect(logs).toEqual([
+            { level: 'DEBUG', message: `Attempting to fulfill API call for Request:${apiCall.id}...` },
+            { level: 'INFO', message: `Submitting API call fulfillment for Request:${apiCall.id}...` },
+          ]);
+          expect(err).toEqual(null);
+          expect(data).toEqual({
+            ...apiCall,
+            fulfillment: { hash: '0xtransactionId' },
+          });
+          expect(staticFulfillMock).toHaveBeenCalledTimes(1);
+          expect(staticFulfillMock).toHaveBeenCalledWith(
```

Comment:
@bdrhn9 These are the `toHaveBeenCalledWith` I meant to get rid of. This is basically repeated everytime and I'd argue it doesn't bring much value.